distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,305 @@
+            ip = DEFAULT_IP;
+            port = DEFAULT_PORT;
+            LOGGER.warn("No IP or port provided.  Using defaults : {}:{}",
+                    DEFAULT_IP,
+                    DEFAULT_PORT);
+        } else if (arguments.length == 2) {
+            ip = arguments[1];
+            port = DEFAULT_PORT;
+            LOGGER.warn("No port provided.  Using default : {}", DEFAULT_PORT);
+        } else {
+            ip = arguments[1];
+            try {
+                port = Integer.parseInt(arguments[2]);
+            } catch (NumberFormatException e) {
+                LOGGER.warn("Unable to parse specified port : {}.  Using Default : {}",
+                        arguments[2],
+                        DEFAULT_PORT);
+                port = DEFAULT_PORT;
+            }
+        }
+
+        LOGGER.info("Video file path : {}", videoFilePath);
+
+        LOGGER.info("Streaming address : {}:{}", ip, port);
+
+        final AtomicLong count = new AtomicLong(0);
+
+        EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+        try {
+
+            Bootstrap bootstrap = new Bootstrap();
+
+            bootstrap.group(eventLoopGroup)
+                    .channel(NioDatagramChannel.class)
+                    .option(ChannelOption.SO_BROADCAST, true)
+                    .handler(new SimpleChannelInboundHandler<DatagramPacket>() {
+                        @Override
+                        protected void channelRead0(ChannelHandlerContext channelHandlerContext,
+                                DatagramPacket datagramPacket) throws Exception {
+                            LOGGER.info("Reading datagram from channel");
+                        }
+
+                        @Override
+                        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
+                                throws Exception {
+                            logErrorMessage(String.format(
+                                    "Exception occured handling datagram packet.  %s",
+                                    cause), false);
+                            ctx.close();
+                        }
+                    });
+
+            Channel ch = bootstrap.bind(0)
+                    .sync()
+                    .channel();
+
+            File videoFile = new File(videoFilePath);
+
+            long bytesSent = 0;
+
+            long tsPacketCount = videoFile.length() / PACKET_SIZE;
+
+            Duration videoDuration = getVideoDuration(videoFilePath);
+            if (videoDuration == null) {
+                return;
+            }
+
+            long tsDurationMillis = videoDuration.toMillis();
+
+            LOGGER.info("Video Duration : {}", tsDurationMillis);
+
+            double delayPerPacket = (double) tsDurationMillis / (double) tsPacketCount;
+
+            long startTime = System.currentTimeMillis();
+
+            int packetsSent = 0;
+
+            try (final FileInputStream fis = new FileInputStream(videoFile)) {
```

Comment:
Wrap this in a `BufferedInputStream` so your calls to `read` aren't literally pushing the pointer one byte at a time through the file. Much faster/more efficient.